### PR TITLE
fix: move concept doc to concept folder

### PR DESCRIPTION
**Pull request by susanshi**
- move concept doc to concept folder ( no change in content)
- update the title of the page to be more consistent

### CODE REVIEW

File: docs/concepts/_category_.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "label": "Concepts",
```
Comment (binbin-li) [inline docs/concepts/_category_.json:2]: not a strong opinion, wonder if `specification` or `architecture` make more sense.

**Review by binbin-li — COMMENTED** [review]
lgtm overall

File: docs/concepts/_category_.json (hunk 1 above)
Comment (susanshi) [reply docs/concepts/_category_.json:2]: good point Binbin.  I want to follow a existing pattern used by other open source docs. E.g. k8s have a concepts (then there is architecture inside the concepts folder). GK docs have both concept and architecture. For Ratify, which content falls under Concept? and which content falls under architecture?

Comment (susanshi) [reply docs/concepts/_category_.json:2]: https://kubernetes.io/docs/concepts/

Comment (binbin-li) [reply docs/concepts/_category_.json:2]: thanks for the reference, I'd like to follow the K8s as well. I feel definition and specification are under concept, while the design and implementation details could fall under architecture. In terms of this PR, I think it's good to keep them under concept.

**Review by binbin-li — APPROVED** [review]

File: docs/concepts/_category_.json (hunk 1 above)
Comment (susanshi) [reply docs/concepts/_category_.json:2]: thanks Binbin. They can be moved around as we add more content

**Pull request merged**